jcl/src/java.base/share/classes/openj9/internal/foreign/abi/InternalDowncallHandler.java

```diff
@@ -689,7 +696,10 @@ private void releaseScope() {
 		for (ResourceScope memArgScope : sessionOrScopeSet) {
 			if (memArgScope.isAlive()) {
 				Thread owner = memArgScope.ownerThread();
-				if (owner == Thread.currentThread()) { // For the confined scope
+				/* The check is intended for the confined scope or
```

Comment:
what happens if one thread releases a scope that is shared?

Reply:
A shared scope is not owned by any thread (that's why there is no owner thread for any shared scope) according to the API Spec. So it won't be released by any thread.